src/UmbracoFileSystemProviders.Azure/AzureBlobFileSystem.cs

```diff
@@ -22,23 +22,53 @@ public class AzureBlobFileSystem : IFileSystem
         /// Initializes a new instance of the <see cref="AzureBlobFileSystem"/> class.
         /// </summary>
         /// <param name="containerName">The container name.</param>
-        /// <param name="rootUrl">The root url.</param>
         /// <param name="connectionString">The connection string.</param>
-        public AzureBlobFileSystem(string containerName, string rootUrl, string connectionString) :
-            this(containerName, rootUrl, connectionString, "365")
+        /// <param name="maxDays">The maximum number of days to cache blob items for in the browser.</param>
+        public AzureBlobFileSystem(string containerName, string connectionString, string maxDays)
         {
+            bool useDevelopmentStorage = false;
+            if (connectionString.Trim().Equals("UseDevelopmentStorage=true", StringComparison.InvariantCultureIgnoreCase))
+            {
+                useDevelopmentStorage = true;
+            }
+
+            this.FileSystem = AzureFileSystem.GetInstance(containerName, connectionString, maxDays, useDevelopmentStorage, false);
```

Comment:
I'm fairly certain this will break everybody's upgrade path if they have chosen to disable the virtual path provider.